### DISCUSSION TITLE
Issues on Mac SE/30

### DISCUSSION THREAD

**Discussion by pgreenland** [ZuluSCSI/ZuluSCSI-firmware#11] [Q&A] [answered]
Hi,

Continued from: https://68kmla.org/bb/index.php?threads/announcing-zuluscsi-a-file-based-scsi-device-emulator.40749/page-7#post-445630

Summary of my problem:

- Updated firmware to 1.0.2
- Loaded a blank disk image
- Initialised disk using a Disk Tools floppy running an unpatched Apple HD SC Setup.
- Installed OS 7.0.1 from floppy successfully.
- Rebooted without floppy present and Mac won't boot

Jumper settings are:

BOOT: Off
TERM: On
DEBUG: Off
SW1: On

SCSI wise, the ZuluSCSI is the only device connected, via the internal connector and a new 15cm ribbon cable (I don't own any other SCSI kit unfortunately).

Booting from a floppy to Disk Tools, the drive is visible and accessible. I'm able to launch applications from it such as a text editor or the control panel widgets. With the Zulu access light flashing briefly during these accesses.

I enabled the jumper switch and saved the log from the SE/30 after powering on without a floppy present.

[zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/8782694/zululog.txt)

Looks like there were a few accesses near the start of the disk, which is what I'd expect from other systems (haven't looked into the boot process much yet).

I attempted to load the image on my OS X powered desktop, renaming to .dmg as suggested by the internet and mounting: 

![image](https://user-images.githubusercontent.com/40036098/170595313-753e48bc-008e-4c49-863b-c2412c41a003.png)

Seems the OS was happy to get it so far and recognised what it claims is a Mac OS Standard volume.

![image](https://user-images.githubusercontent.com/40036098/170595460-7e86ad82-8b49-4576-afc3-1c2da9bae3ae.png)

Opening the image in a hex editor, while not super useful, there's a few strings standing out Apple_partition_map and Apple_HFS within the first few KB, followed by a large blank area, before presumably a partition starts.

Aside from the lack of booting it seems to be working really well.

Thoughts would be appreciated.

Thanks,

Phil

**Comment by PetteriAimonen** [comment]
Hmm, we are investigating an issue with Mac Plus not booting from ZuluSCSI, but I'm not sure if the same problems might apply to Mac SE/30.

You can try the attached test firmware and set these settings in `zuluscsi.ini`:
[zuluscsi_mac_test.zip](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/8786142/zuluscsi_mac_test.zip)

```
[SCSI]
SelectionDelay = 0
Quirks = 1
EnableUnitAttention = 0
EnableSelLatch = 1
```

**Reply by pgreenland** [reply]
Hi,

Given the updated firmware a go with the config file.

Here's the debug output, during a failed boot:

[zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/8786469/zululog.txt)

Still no booting from SCSI unfortunately.

Booted into disk tools again with the firmware and setting applied, disk is still accessible looking good.

I attempted to get the disk image generated by the ZuluSCSI to mount in Mini vMac but it doesn't seem to support hard disk images. The idea being to see if the image was good and booting was an issue or if the image was somehow bad and booting wasn't the problem. Would it be worth you dropping me a small image that boots on your SE to try?

Thanks,

Phil

**Reply by PetteriAimonen** [reply]
Based on log it appears to work the same as earlier, so this issue is probably different than the Mac Plus one.

@aperezbios: do you have a working Mac SE image?

**Reply by pgreenland** [reply]
Progress update from my end.

Built Basilisk II which happily boots the image generated by the ZuluSCSI from my initial install from floppy disk.

Although I wasn't able to get SCSI emulation working so I'm not sure how much of the image actually ends up in the emulated environment. Installing onto a blank image for example results in an image without a partition table. So wondering if Basilisk reads and skips the partition table itself as its not really emulating the SCSI disk when it comes to its "volumes".

I also tried a spare ROM in the SE/30, from what I believe to be a later model logic board. I believe (but don't know for sure) that my logic board is a fairly early one. Its got a socketed CPU and what I thought was a repair but may be a factory bodge wire between a chip (UI6) and a test point.

Either way still not booting sadly :-(

**Comment by pgreenland** [comment, marked as answer]
@PetteriAimonen solved my problem. I thought I'd try formatting with a different utility. So tried Lido 7.56, stuffed into a System 6.0.8 Disk Tools floppy. Installed System 6 to the freshly formatted disk and booted straight away.

From what I read and from poking around in the images. After the Apple Partition table there's an Apple_Driver partition, which contains some sort of SCSI device driver.

The ROM makes use of the driver somehow, to access the disk, on route to booting the OS.

Presumably the driver installed by Apple HD SC Setup takes exception to the ZuluSCSI for some reason on the SE/30 at least.

I've generated several images, initially formatting with Lido and installing. But now just baking filesystem images, generating a partition table before injecting the lido driver partition and a HFS partition from an emulator.

All working well.

Would be nice to know why the standard Apple tools aren't working, but thats for another day.

Performance wise Lido reported that I'm getting 1807 Kbytes/sec read and 1248 Kbytes/sec write performance with an average access time of 9ms :-P.

**Reply by PetteriAimonen** [reply]
Very useful information! Compatibility with the Apple driver is certainly something we want to fix, perhaps this will help resolve the differences between the two.

**Reply by pgreenland** [reply]
Let me know if you need a hand testing / debugging when the time is right. I wouldn't know where to start but looking forward to finding out.

**Reply by aperezbios** [reply]
@pgreenland for the access times, those seem rather high, _unless_ you have debug enabled, in which case it's expected, as the firmware has to interrupt reads and writes from the OS to write from the in-memory log buffer to the log on the SD card itself.

Double-check that debug is disabled. if it wasn't, you should see the latency decrease significantly if you set debug=0, or comment out the debug line in your ini file.
